tests/test_app.py

```diff
@@ -52,3 +54,43 @@ def test_rollbacks(self):
             response = client.get(f"/{self.subtype_id}")
             # Expect a 404 status code, indicating that the subtype was not found
             self.assertEqual(response.status_code, 404)
+
+    @patch("app.main.extract_data")
+    def test_analyze_document(self, mock_extract_data):
+        mock_inspection_data = {
+            "company_name": "Test Company",
+            "fertiliser_name": "Mock Fertilizer",
+            "registration_number": "REG123",
+        }
+        mock_inspection = FertilizerInspection.model_validate(mock_inspection_data)
+        mock_extract_data.return_value = mock_inspection
+
+        # Mock files for testing
+        file_content_1 = b"Sample content 1"
+        file_content_2 = b"Sample content 2"
+
+        files = [
+            ("files", ("file1.txt", file_content_1, "text/plain")),
+            ("files", ("file2.txt", file_content_2, "text/plain")),
+        ]
+
+        with TestClient(app) as client:
+            response = client.post("/analyze", files=files)
```

Comment:
Need more test cases?
Empty file
file list with one or many empty files
empty file list

Reply:
done. I now raise 422 in those cases in the route.